Sources/App/Commands/VersionReAnalyzer.swift

```diff
@@ -0,0 +1,236 @@
+import Vapor
+import Fluent
+
+
+struct ReAnalyzeVersionsCommand: Command {
+    let defaultLimit = 1
+
+    struct Signature: CommandSignature {
+        @Option(name: "limit", short: "l")
+        var limit: Int?
+        @Option(name: "id")
+        var id: UUID?
+        @Option(name: "before")
+        var before: Date?
+    }
+
+    var help: String { "Run version re-analysis" }
+
+    func run(using context: CommandContext, signature: Signature) throws {
+        let limit = signature.limit ?? defaultLimit
+
+        let client = context.application.client
+        let db = context.application.db
+        let logger = Logger(component: "re-analyze-versions")
+        let threadPool = context.application.threadPool
+
+        guard let cutoffDate = signature.before ?? Current.reAnalyzeVersionsBeforeDate() else {
+            logger.info("No cut-off date set, skipping re-analysis")
+            return
```

Comment:
This is a potential mechanism to later deploy this as an automatic command that runs on a cycle like the others. By setting `RE_ANALYZE_VERSIONS_BEFORE_DATE` to a `YYYY-MM-DD` value, we'll start processing those versions until they all have been updated.

We can leave the job running and simple set the value to `-` (or anything that isn't `YYYY-MM-DD`) or remove the key to make the process a no-op.

Alternatively, we can kick the process off manually whenever we need to without setting up an automatic process and the env variable.